platform/os_windows.go

```diff
@@ -294,6 +292,50 @@ func SetSdnRemoteArpMacAddress(execClient ExecClient) error {
 	return nil
 }
 
+// straight out of chat gpt
+func restartService(serviceName string) error {
+	// Connect to the service manager
+	m, err := mgr.Connect()
+	if err != nil {
+		return fmt.Errorf("could not connect to service manager: %v", err)
+	}
+	defer m.Disconnect()
+
+	// Open the service by name
+	service, err := m.OpenService(serviceName)
+	if err != nil {
+		return fmt.Errorf("could not access service: %v", err)
+	}
+	defer service.Close()
+
+	// Stop the service
+	_, err = service.Control(svc.Stop)
+	if err != nil {
+		return fmt.Errorf("could not stop service: %v", err)
+	}
+
+	// Wait for the service to stop
+	status, err := service.Query()
+	if err != nil {
+		return fmt.Errorf("could not query service status: %v", err)
+	}
+	for status.State != svc.Stopped {
+		time.Sleep(500 * time.Millisecond)
+		status, err = service.Query()
+		if err != nil {
+			return fmt.Errorf("could not query service status: %v", err)
+		}
+	}
```

Comment:
This should consult a `context.Context` for cancelation so that it will not run indefinitely.